mood-analysis.rb

```diff
@@ -19,6 +19,52 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def strip_punctuation(sentence)
+  sentence.gsub(/[[:punct:]]/, "")
+end
+
+def happy_days(journal)
+  entries = 0
+  happy_count = 0
+  # wanted to do this with reduce but didn't get it figuerd out :/
+  journal.each do |entry|
+    happy_count += 1 if analyze_mood(strip_punctuation(entry)) == ":-)"
+    entries += 1
+    if happy_count > 3
+      break
+    end
```

Comment:
Also, this could be reduced by inserting the return into the if. There's no advantage of breaking and then testing the condition again later, when we can just make the return right here.